### PR TITLE
Fix references resolving

### LINKED ISSUE #58: Parse error must emit build warning
Currently it seems all wrong input is silently accepted and all internal parse errors supressed.

Example rst:
```
.. php:class:: Scope

.. php:method:: simplify();
```

notice the `;` at the method end. The `;` should not be present, but when present, the rendered result is very off:

```diff
 <dl class="php method">
-<dt class="sig sig-object php" id="Scope::simplify">
-<span class="sig-prename descclassname"><span class="pre">Scope::</span></span>
-<span class="sig-name descname"><span class="pre">simplify</span></span>
-<span class="sig-paren">(</span><span class="sig-paren">)</span>
-<a class="headerlink" href="#Scope::simplify" title="Permalink to this definition">¶</a></dt>
+<dt class="sig sig-object php">
+<span class="sig-name descname"><span class="pre">simplify();</span></span></dt>
 <dd></dd></dl>
```

catching such typos manually is very hard.

Sphinx does support build warnings, example:

image

So I expect this extension to emit warning on wrong input as well.

Zero warnings can be then then easily asserted using CI.

### PR DESCRIPTION
**Pull request by mvorisek**
fix part of #56 - fixes absolute references (but declarations still cannot currently contain `\`)

also fix #58

also missing targets are newly reported, as infos only, as some targets might be in other projects, global PHP classes, ... and CI must pass

also the testing is unified and the html output for all original tests is asserted using CI

also the code is reformatted, in separate commit, using Black to enforce strict coding style

output of all rst test files is exactly the same as before this PR -> no BC break

### CODE REVIEW

File: test/unit/ns.html [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+      </dt>
+      <dd/>
+    </dl>
+    <section id="id1">
+      <h2>Cross linking<a class="headerlink" href="#id1" title="Permalink to this heading">&#xB6;</a></h2>
+      <ul class="simple">
+        <li>
+          <p>
+            <code class="xref php php-meth docutils literal notranslate">
+              <span class="pre">\Foo\Bar::simplify</span>
```
Comment (mvorisek) [inline test/unit/ns.html:118]: TODO

**Comment by mvorisek** [comment]
@markstory PR is complete, it focuses to support `\OtherNs\Cl::method()` references which are important to be supported when the targets are outside the current NS. No BC break should be involved.

**Comment by markstory** [comment]
This is a big change, it might take me a few days to get through it.

File: test/rst_doc.md [hunk 2]
```diff
@@ -1,3 +1,4 @@
+```{eval-rst}
```
Comment (markstory) [inline test/rst_doc.md:1]: Is this to simplify build configuration? Having this markdown/rst hybrid format be the default isn't ideal to me. It makes a less-frequently used package a hard dependency and requirement for long term maintenance.

File: test/rst_doc.md [hunk 3]
```diff
@@ -468,27 +473,27 @@ Return Types
...
     .. php:method:: returnClassConstant()
 
         :returns: The value of a specific class constant.
-        :returntype: LibraryName\\NamespaceClass::NAMESPACE_CONST
+        :returntype: \\LibraryName\\NamespaceClass::NAMESPACE_CONST
 
     .. php:method:: returnGlobalConstant()
 
-        :returns: The value of a specific global constant.
+        :returns: The value of a specific global constant. # TODO link is not working without "\\"
```
Comment (markstory) [inline test/rst_doc.md:490]: What's up here?

File: test/rst_doc.md (hunk 2 above)
Comment (mvorisek) [reply test/rst_doc.md:1]: Sphinx requires only one file format. To allow to test everything at once, we need md files + `{eval-rst}`.

- MyST parser needs to be tested to assert the parsing is done solely by Sphinx and the phpdomain does not operate on raw rst. This is related with #55 (I plan to fix it in another PR).

- `{eval-rst}` seems to work perfectly as native rst, Sphinx simply does not read rst file directly, but instead the wrapped string. I did not experience any difference between native rst file and rst from evaluated string.

MyST parser is very popular and the dev/test dependency should not be a problem.

File: test/rst_doc.md (hunk 3 above)
Comment (mvorisek) [reply test/rst_doc.md:490]: `:returntype: SOME_CONSTANT` return type reference with constant is unresolved (has no clickable link) - I need to find a solution first. If ok, let's merge this big PR and these small TODOs can be fixed later, probably together with #56.

**Comment by mvorisek** [comment]
@markstory the changes are not smaller, but the most of the changed LoC are changes from generated expected html and changes due Black CS. If you want the the Black CS changes as separate PR, let me know, but as I see you merge PRs using merge, thus the individual commits will be preserved even if merged as one PR.

File: test/rst_doc.md (hunk 2 above)
Comment (markstory) [reply test/rst_doc.md:1]: > MyST parser is very popular and the dev/test dependency should not be a problem.

Fair enough, this is the first time I've encountered it.

**Comment by markstory** [comment]
I had to turn off tests for python 3.9+ as they started failing. I've also turned off the cron schedule for CI as it has been failing for a few days, likely due to a change in sphinx.

**Pull request merged**